src/core/alloc.h

```diff
@@ -37,6 +37,23 @@ MVM_STATIC_INLINE void * MVM_realloc(void *p, size_t size) {
     return ptr;
 }
 
+/* This doesn't use any mimalloc functions because it's for use in NativeCall
+ * code, where modules may be freeing or allocating memory and wouldn't be 
+ * using mimalloc. */
+MVM_STATIC_INLINE void * recalloc(void *p, size_t old_size, size_t size) {
+    void *ptr = realloc(p, size);
```

Comment:
I think this code is only used in one place, and I think that we're unlikely to need it in a second place. And it's a non-standard "API" where `mimalloc` and `MVM` disagree about the parameters. So I'd be tempted to inline this into its callsite.

**but**

I'd be happy to do that as a later commit. I don't want to delay fixing the real regression. I was only adding this comment on the basis that I was trying to write a proper review, and I didn't want to let the perfect be the enemy of the good.

Reply:
> I think this code is only used in one place, and I think that we're unlikely to need it in a second place. And it's a
> non-standard "API" where mimalloc and MVM disagree about the parameters. So I'd be tempted to inline this into
> its callsite.

Yeah, I hit the need for this pretty soon and just assumed I'd need it again. I'm 50/50 on this. Anyone else have an opinion?